File: leetcode/algorithms/283. 移动零/2.java
class Solution {
    public void moveZeroes(int[] nums) {
        int len = nums.length;
        int j=0,i;
        for(i=0;i<len;i++){
            if(nums[i]!=0){
                nums[j] = nums[i];
                j++;
            }
        }
        while (j<len){
            nums[j] =0;
            j++;
        }
    }
}

File: leetcode/algorithms/53. 最大子数组和/2.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int count = 0;
        int result = INT32_MIN;
        for (int i = 0; i < nums.size(); ++i) {
            count+=nums[i];
            if(count>result){
                result = count;
            }
            if(count<=0){
                count = 0;
            }
        }
        return result;
    }
};
int main()
{
    vector<int> num = {1,-17,5,10,13,15,10,5,16,8};
    cout<<(new Solution)->maxSubArray(num)<<endl;
}
